lib/jekyll/renderer.rb

```diff
@@ -135,10 +135,13 @@ def place_in_layouts(content, payload, info)
 
       used   = Set.new([layout])
 
+      # Reset the payload layout data to ensure it starts fresh for each page.
+      payload["layout"] = nil
+
       while layout
         payload['content'] = output
         payload['page']    = document.to_liquid
-        payload['layout']  = Utils.deep_merge_hashes(payload['layout'] || {}, layout.data)
+        payload['layout']  = Utils.deep_merge_hashes(layout.data, payload["layout"] || {})
```

Comment:
@benbalter This is a breaking change that is considered a bug fix (it was a bug that the default layout, for example, overwrote the metadata from a more specific/child layout like "page" or "post). How can I best communicate this change?